app/services/api/pipeline_api.rb

```diff
@@ -136,10 +133,7 @@ def check_samples_status
         body: tweets
       )
     else
-      Helpers::ApiResponse.new(
-        status: :success,
-        body: tweets
-      )
+      Helpers::ApiResponse.new(status: :success, body: [tweets, s3_keys])
```

Comment:
It has just one user, `check_samples_status` rake task. The idea is that if `Setting.sampled_status` is rewritten inside the services, then once the rake runs unsuccessfully once, it cannot access the latest sample on S3 (since it get indicated as processed by rewriting `Setting.sampled_status`), which I fix by passing the keys to rewrite them upon success of the rake task.